portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -89,21 +89,17 @@ private String getStringParameter(HttpServletRequest request, String name, Strin
 
   /** Get number of comments to display from request */ 
   private int getNumCommentsParameter(HttpServletRequest request) {
-    String numCommentsString = request.getParameter(NUM_COMMENTS_PARAMETER);
+    String numCommentsString = request.getStringParameter(request, NUM_COMMENTS_PARAMETER, "");
 
     // Convert string value to int
     int numComments;
     try {
       numComments = Integer.parseInt(numCommentsString);
```

Comment:
Assume the parameter read has the empty string as the default value (you're doing that already). Then here, you could use another ternary and you wouldn't need a return statement at the end of the method:

```
try {
  return numCommentsString.isEmpty() : defaultValue ? Integer.parseInt(numCommentsString); 
}
catch (NumberFormatException e) {
  System.out.println(...);
  return Integer.MIN_VALUE;
}
```

For code understanding: There is no flow once you get past the try/catch block where you could run any code - you've already returned no matter what.

Reply:
This still needs work